runtime/src/main/java/com/dylibso/chicory/runtime/ByteBufferMemory.java

```diff
@@ -0,0 +1,394 @@
+package com.dylibso.chicory.runtime;
+
+import static com.dylibso.chicory.runtime.ConstantEvaluators.computeConstantValue;
+import static java.lang.Math.min;
+
+import com.dylibso.chicory.wasm.ChicoryException;
+import com.dylibso.chicory.wasm.UninstantiableException;
+import com.dylibso.chicory.wasm.types.ActiveDataSegment;
+import com.dylibso.chicory.wasm.types.DataSegment;
+import com.dylibso.chicory.wasm.types.MemoryLimits;
+import com.dylibso.chicory.wasm.types.PassiveDataSegment;
+import java.nio.BufferOverflowException;
+import java.nio.BufferUnderflowException;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+
+/**
+ * Represents the linear memory in the Wasm program. Can be shared
+ * reference b/w the host and the guest.
+ */
+public class ByteBufferMemory implements Memory {
```

Comment:
Make this final?

Reply:
This should be the only "open" discussion, I'm not 100% convinced we should make it `final` as it seem appealing to just extend instead of having to spin up a full-blown implementation all the time.
But happy to listen to a different POV!